tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge_wrapper/signup_info_kme.cpp

```diff
@@ -60,16 +69,48 @@ std::map<std::string, std::string> CreateEnclaveDataKME(
     result["enclave_quote"] = enclave_quote;
 
     return result;
-}
+}  // SignupInfoKME::CreateEnclaveData
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+std::map<std::string, std::string> SignupInfoKME::UnsealEnclaveData() {
+    tcf_err_t presult;
+
+    // UnsealEnclaveData will resize appropriately
+    StringArray public_enclave_data(0);
+
+    SignupDataKME signup_data;
+    presult = signup_data.UnsealEnclaveData(
+        public_enclave_data);
+    ThrowTCFError(presult);
+
+    // Parse the json and save the verifying and encryption keys
+    std::string verifying_key;
+    std::string encryption_key;
+    std::string encryption_key_signature;
+
+    presult = SignupInfo::DeserializePublicEnclaveData(
+        public_enclave_data.str(),
+        verifying_key,
+        encryption_key,
+        encryption_key_signature);
+    ThrowTCFError(presult);
+
+    std::map<std::string, std::string> result;
+    result["verifying_key"] = verifying_key;
+    result["encryption_key"] = encryption_key;
+    result["encryption_key_signature"] = encryption_key_signature;
+
+    return result;
+}  // SignupInfoKME::UnsealEnclaveData
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-size_t VerifyEnclaveInfoKME(
+size_t SignupInfoKME::VerifyEnclaveInfo(
     const std::string& enclave_info,
     const std::string& mr_enclave,
     const std::string& ext_data) {
 
-    tcf_err_t result = tcf::enclave_api::enclave_data_kme::VerifyEnclaveInfoKME(
+    SignupDataKME signup_data;
+    tcf_err_t result = signup_data.VerifyEnclaveInfo(
         enclave_info, mr_enclave, ext_data);
-    size_t verify_status = result;
-    return verify_status;
+    return (size_t) verify_status;
```

Comment:
Doesn't looks like `verify_status` is defined ? Are these files getting compiled ? These files should probably be included in swig interface file `avalon_enclave.i` (https://github.com/hyperledger/avalon/blob/master/enclave_manager/avalon_enclave_manager/avalon_enclave.i) and be verified for compilation. Even better option is as part of signup_info refactoring, better to create 3 interface files - `avalon_enclave_singleton.i`, `avalon_enclave_kme.i`, `avalon_enclave_wpe.i` .

Reply:
I missed to add this cpp file in setup.py to get it compiled, so i didn't notice this. I will correct it.
Also, i will include the cpp file as part of setup.py